modules/payments/modules/payments/paymentHistoryModal.js

```diff
@@ -0,0 +1,130 @@
+import React, { PureComponent, useState, useEffect } from 'react';
+import { View, Text, Modal, StyleSheet, Pressable, Alert, RefreshControl } from 'react-native';
+import { FlatList, TouchableOpacity } from 'react-native-gesture-handler';
+import { fetchPaymentHistory } from "./api";
+import { Dimensions } from 'react-native'
+let deviceWidth = Dimensions.get('window').width
+
+
+const PaymentHistoryModal = (props) => {
+
+    const [paymentHistory, setPaymentHistory] = useState([]);
+    const [modalVisible, setModalVisible] = useState(false);
+    const [refresh, setRefresh] = useState(false);
+
+    useEffect(() => {
+        get_payments();
+    }, [])
+
+    const get_payments = async () => {
+        setRefresh(true);
+        const res = await fetchPaymentHistory()
+        fetchPaymentHistory().then(res => {
+            setRefresh(false);
+            setPaymentHistory(res);
+        })
+    }
+
+    const renderItem = ({ item }) => {
+        return (
+            <View style={styles.listItemContainer}>
+                <Text>{item.amount} cents {item.currency}</Text>
+                <Text>
+                    <Text style={styles.bold}>Payment Method:</Text> 
+                    {item?.payment_method_details?.card?.brand} - 
+                    {item?.payment_method_details?.card?.last4} - {item?.payment_method_details?.card?.wallet?.type}
+                </Text>
+                <Text><Text style={styles.bold}>Status:</Text> {item.status}</Text>
+            </View>
+        )
+    }
```

Comment:
You can null check `item` at first, avoiding having to conditionally chain it in the JSX.
```suggestion
    const renderItem = ({ item }) => {
        if (!item) return;
        return (
            <View style={styles.listItemContainer}>
                <Text>{item.amount} cents {item.currency}</Text>
                <Text>
                    <Text style={styles.bold}>Payment Method:</Text> 
                    {item.payment_method_details?.card?.brand} - 
                    {item.payment_method_details?.card?.last4} - {item.payment_method_details?.card?.wallet?.type}
                </Text>
                <Text><Text style={styles.bold}>Status:</Text> {item.status}</Text>
            </View>
        )
    }
```